src/test/java/com/epam/test_generator/dao/SuitDAOTest.java

```diff
@@ -136,9 +136,75 @@ public void testUpdateName() {
         Assert.assertEquals(newSuit, suitDAO.getOne(id));
     }
 
+    @Test
+    public void testUpdateWithRowNumbers() {
+
+        final List<Suit> expectedSuits = suitDAO.save(getExpectedSuitsFromDatabase());
+        final List<SuitRowNumberUpdate> sortedRowNumbers = getSortedSuitsRowNumbers(expectedSuits);
+
+        final List<Long> ids = sortedRowNumbers
+                .stream()
+                .map(SuitRowNumberUpdate::getId)
+                .collect(Collectors.toList());
+
+        final List<Suit> actualSuits = suitDAO.findByIdIn(ids, new Sort(Sort.Direction.ASC, "id"));
+        Assert.assertEquals(expectedSuits, actualSuits);
+
+        final List<Suit> updatedSuits = updateByRowNumbers(actualSuits, sortedRowNumbers);
+        final List<Suit> suitsAfterChange = suitDAO.save(updatedSuits);
+        Assert.assertEquals(updatedSuits, suitsAfterChange);
+    }
+
+    private List<Suit> updateByRowNumbers(List<Suit> actualSuits, List<SuitRowNumberUpdate> sortedRowNumbers) {
+        for (int i = 0; i < actualSuits.size(); i++) {
+            final Integer newRowNumber = sortedRowNumbers.get(i).getRowNumber();
+            actualSuits.get(i).setRowNumber(newRowNumber);
+        }
+        return actualSuits;
+    }
+
+    private List<SuitRowNumberUpdate> getSortedSuitsRowNumbers(List<Suit> suits) {
+        List<Integer> rowNumbers = suits.stream().map(Suit::getRowNumber).collect(Collectors.toList());
+        Collections.shuffle(rowNumbers);
+
+        List<Suit> updateSuits = new ArrayList<>();
+        for (int i = 0; i < suits.size(); i++) {
+            Suit suit = suits.get(i);
+            updateSuits.add(new Suit(
+                    suit.getId(),
+                    suit.getName(),
+                    suit.getDescription(),
+                    suit.getCases(),
+                    suit.getPriority(),
+                    suit.getTags(),
+                    suit.getRowNumber())
+            );
+            updateSuits.get(i).setRowNumber(rowNumbers.get(i));
+        }
+
+        return updateSuits
+                .stream().map(suit -> getSuitRowNumberUpdateInstance(suit.getId(), suit.getRowNumber())).collect(Collectors.toList())
+                .stream().sorted(Comparator.comparingLong(SuitRowNumberUpdate::getId)).collect(Collectors.toList());
+    }
+
+    private SuitRowNumberUpdate getSuitRowNumberUpdateInstance(Long id, int rowNumber) {
```

Comment:
Remove methods getSuitRowNumberUpdateInstance, getExpectedSuitsFromDatabase, getSortedSuitsRowNumbers from test class. These methods do not improve code readability, nor help to avoid code duplication. It is not shame to hardcode expected result in test body.